src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/OutOfProcTagHelperResolver.cs

```diff
@@ -18,21 +19,17 @@
 
 namespace Microsoft.CodeAnalysis.Remote.Razor;
 
-internal class OOPTagHelperResolver : ITagHelperResolver
+[Export(typeof(ITagHelperResolver))]
```

Comment:
I think I need to put a post-it on my screen to remind me which namespace the `Export` attribute has to come from to make the lack of `Shared` allowed 😛

Reply:
We should just ban the different symbols by layer. Roslyn does this for the Features layer: https://github.com/dotnet/roslyn/blob/bf08f8ee87d606007d41997e149b1166ad26a2b5/src/Features/BannedSymbols.txt#L4.

Reply:
I thought there was some occasions that called for v1, like `AllowDefault` or something, but that's probably just a lack of knowledge about what the v2 method for that is.

Reply:
Wait... that file bans `System.ComponentModel.Composition`, which is what you're using here. So does that mean this _should_ have a `Shared`?

Reply:
Errr... no. `[Shared]` is only used by MEFv2. This is exported as MEFv1 because it is only used in VS.

Note: A lot of the whole MEFv1, MEFv2 business shouldn't really exist in Razor. It's there because Razor tooling has gotten very dirty about defining MEF exports in lower layers that really shouldn't be there, since they aren't needed in the language server.

Reply:
> Wait... that file bans `System.ComponentModel.Composition`, which is what you're using here. So does that mean this _should_ have a `Shared`?

I was not suggesting that we just take that file wholesale. I was suggesting that once the Roslyn tooling layers are properly defined, we should have a _strategy_ of using a `BannedSymbols.txt` file to help folks do the right things. IMO, Razor tooling should _never_ use MEFv2 in its current architecture. It does because there a lot of layering violations and because it used Roslyn's workspace/language service infrastructure. Of course, co-hosting will likely change that architecture, but we need to get things clean to reduce the confusion.